tests/test_early_stopping.py

```diff
@@ -44,10 +44,11 @@ def test_is_improvement():
 class MockEvaluator(Evaluator):
     """A mock evaluator for testing early stopping."""
 
-    def __init__(self, losses: Iterable[float]) -> None:
+    def __init__(self, losses: Iterable[float], automatic_memory_optimization: bool = True) -> None:
         super().__init__()
```

Comment:
incorrect call to super?

Reply:
should pass automatic_memory_optimization to `super().__init__` instead of setting in the class

Reply:
Done in 9a6a26f0086974c9aeb924a4d0187d424664d23b.

Reply:
double check this one

Reply:
af20d329